File: Frontend/src/domain/Types/entities/StockItem.ts
import type { StorageArea } from "./StorageArea";

export type StockItem = {
  containerISO: string;
  description: string;
  from: string;
  to: string;
  availableSince: string;
  availableUntil: string;
  storageArea: StorageArea;
};

File: JWP_API/JadeWesserPort/Domain/Entities/StockItem.cs
using Microsoft.EntityFrameworkCore;

namespace JadeWesserPort.Domain.Entities;

[Index(nameof(ContainerISO), IsUnique = true)]
public class StockItem
{
    public Guid Id { get; set; }
    public string ContainerISO { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string From { get; set; } = string.Empty;
    public string To { get; set; } = string.Empty;
    public DateTime AvailableSince { get; set; }
    public DateTime AvailableUntil { get; set; }

    public Guid StorageAreaId { get; set; }
    public StorageArea StorageArea { get; set; } = null!;
}
